lib/measured/rails/validations.rb

```diff
@@ -6,13 +14,40 @@ def validate_each(record, attribute, measurable)
     measurable_unit = record.send("#{ attribute }_unit")
     measurable_value = record.send("#{ attribute }_value")
 
-    message = options[:message] || "is not a valid unit"
+    return unless measurable_unit.present? || measurable_value.present?
+
+    record.errors.add(attribute, message) if [measurable_unit.blank?, measurable_value.blank?].any?
+
+    record.errors.add(attribute, message) unless measured_class.valid_unit?(measurable_unit)
 
     if options[:units]
       valid_units = [options[:units]].flatten.map{|u| measured_class.conversion.to_unit_name(u) }
       record.errors.add(attribute, message) unless valid_units.include?(measured_class.conversion.to_unit_name(measurable_unit))
     end
 
-    record.errors.add(attribute, message) unless measured_class.valid_unit?(measurable_unit)
+    options.slice(*CHECKS.keys).each do |option, value|
+      record.errors.add(attribute, message) unless measurable.send(CHECKS[option], value_for(value, record))
+    end
+  end
+
+  private
+
+  def message
+    options[:message] || "is not a valid unit"
+  end
+
+  def value_for(key, record)
+    value = case key
+    when Proc
+      key.call(record)
+    when Symbol
+      record.send(key)
+    else
+      key
+    end
+
+    raise ArgumentError, ":#{ value } must be a Measurable object" unless value.is_a?(Measured::Measurable)
```

Comment:
I guess we could do both, but that seems unnecessary. Though it would warn us when a scalar number us used, which is the main thing we will need to warn against.

```ruby
validates :length, measured: { greater_than: 10 } # this makes no sense because 10 what?
```